File: programmers/lv2/택배상자.py
def solution(order):
    answer = 0
    sub = []
    
    i = 1
    while i < len(order) + 1:
        sub.append(i)
        while sub and sub[-1] == order[answer]:
            answer += 1
            sub.pop()
        i += 1
    
    return answer

File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[] order) {
        int answer = 0;
        Stack<Integer> sub = new Stack<>();        
            
        int index = 0;
        for(int i = 1 ; i <= order.length; i++){
            if (order[index] == i) {
                index++;
                answer++;
            } else {
                sub.push(i);
            }
            while (!sub.isEmpty() && sub.peek() == order[index]) {
                sub.pop();
                index++;
                answer++;
            }
        }

        return answer;
    }
}